### PR TITLE
Backport PR #1745 to "maint/7.2" branch

### PR DESCRIPTION
**Pull request by rahulporuri**
This PR backports the PR #1745 to the `maint/7.2` branch in preparation for the 7.2.2 bugfix release. This PR also updates the changelog in preparation for the release.

**Checklist**
- [ ] ~Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)~

### CODE REVIEW

**Review by mdickinson — APPROVED** [review]
LGTM as a backport, but why is there no regression test in the original fix?

**Comment by rahulporuri** [comment]
> LGTM as a backport, but why is there no regression test in the original fix?

I'll try to add a regression test.

File: traitsui/wx/tree_editor.py [hunk 1]
```diff
@@ -1256,9 +1256,12 @@ def _new_actions(self, node, object):
                 name = class_name
             if not factory:
                 factory = klass
-            def perform_add(object):
+            def perform_add(object, factory, prompt):
                 self._menu_new_node(factory, prompt)
-            items.append(Action(name=name, on_perform=perform_add))
+            on_perform = partial(
```
Comment (mdickinson) [inline traitsui/wx/tree_editor.py:1261]: `partial` is undefined here (and in the original PR that this was backported from)

**Comment by corranwebster** [comment]
Closing as this is no longer needed.

**Pull request closed without merging**